mempool/src/interface/mempool_interface_impl/pool/tx_with_fee.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::{chain::signed_transaction::SignedTransaction, primitives::Amount};
+
+use crate::{error::TxValidationError, get_memory_usage::GetMemoryUsage};
+
+use super::try_get_fee::TryGetFee;
+use super::Mempool;
+
+pub struct TxWithFee {
+    tx: SignedTransaction,
+    fee: Amount,
+}
+
+impl TxWithFee {
+    pub async fn new<M: GetMemoryUsage + Sync + Send>(
+        mempool: &Mempool<M>,
+        tx: &SignedTransaction,
+    ) -> Result<Self, TxValidationError> {
+        let fee = mempool.try_get_fee(tx).await?;
+        Ok(Self {
+            tx: tx.clone(),
```

Comment:
I think this method should take ownership of `tx` and let the caller handle the potential cloning. If the caller already owns the `tx` and does not need it for anything else, it can be just passed in without cloning.

Reply:
Done.